### DISCUSSION TITLE
Idea: Showing split chapters as one gallery with a dropdown menu

### DISCUSSION THREAD

**Discussion by ghost** [Difegue/LANraragi#356] [Ideas]
All of my split chapters from books ([example](https://i.imgur.com/EeY143i.png)) are showing up as separate galleries. 
My idea around this problem is to treat every folder as "one gallery" if it has "@@" in the beginning of the name. 
If LANraragi detects a folder with "@@" in the beginning of its name, it'll treat it as one gallery. 

Is this a dumb or good idea?

**Comment by Difegue** [comment]
Well, I wouldn't implement it this way, but there's been a few requests to allow the reader to automatically read multiple archives in a row.

One of the user survey requests was to allow reading through all archives in a category one after each other -- Alongside the subfolder to categories script, this would fix your issue, I think.

**Comment by ghost** [comment]
Wouldn't that only work if the user has one folder with only zip files and subfolders in a very strict way? That seems unpractical for people like me with a sorted directory tree and many files/folders. 
```
Storage/Gallery_with_split_chapters\splitchapters.zip
Storage/Gallery.zip
```

Another solution to the problem with split chapters is to create a text file in a folder with split chapters and give it a unique name and have LANraragi treat any folder as one gallery if it finds a text file with such a name.
Assume there is a text file called @@ in "Gallery_with_split_chapters", this means that those folders and anything underneath it is one entire gallery, and in LANraragi, it shows up as 3 galleries (Gallery231, Gallery100 and Gallery90". I think this approach is slightly better than my first idea.
```
Storage/RandomGroup/Artist342984/Gallery231_with_split_chapters/
Storage/Artist35/English/Galleries100_with_split_chapters/
Storage/Group249/English/Gallery90.zip
```

**Comment by ghost** [comment]
[Storage for LANraragi.zip](https://github.com/Difegue/LANraragi/files/5435947/Storage.for.LANraragi.zip)

There are 8 galleries and 16 zip files in this mock-up.
LANraragi in its current state will present those 16 zip files as 16 galleries, but with my idea, it will only show 8 galleries, and the folder "@@ gallery8" will show up as "gallery8" with a dropdown menu with whatever is inside the folder. Hope this clears up any misunderstandings.
